go/src/vendors/sonic/client/dbclient/counter_repository.go

```diff
@@ -0,0 +1,75 @@
+// Defines APIs to get interface counters from redis db - CountersDB.
+package dbclient
+
+import (
+	"strconv"
+	. "go/src/lib/lomcommon"
+)
+
+const (
+	counters_port_name_map_redis_key      string = "COUNTERS_PORT_NAME_MAP"
+	counters_db_table_name                string = "COUNTERS:"
+	sai_port_stat_if_in_errors_field      string = "SAI_PORT_STAT_IF_IN_ERRORS"
+	sai_port_stat_if_in_ucast_pkts_field  string = "SAI_PORT_STAT_IF_IN_UCAST_PKTS"
+	sai_port_stat_if_out_ucast_pkts_field string = "SAI_PORT_STAT_IF_OUT_UCAST_PKTS"
+	if_in_errors_counter_key              string = "IfInErrors"
+	in_unicast_packets_counter_key        string = "InUnicastPackets"
+	out_unicast_packets_counter_key       string = "OutUnicastPackets"
+	atoi_base                                int = 10
+        uint_bit_size                            int = 64
+)
+
+type CounterRepository struct {
+	RedisProvider RedisProviderInterface
+}
+
+/* Cache for storing interface to Oid mapping */
+var interfaceToOidMapping map[string]string
+
+/*
+Returns interface counters for all interfaces on the Sonic device.
+First it gets all oids for interfaces and then gets counters for each interface by performing redis hmGet calls.
+*/
+func (counterRepository *CounterRepository) GetInterfaceCounters() (map[string]map[string]uint64, error) {
```

Comment:
Suggestion. You can assign type as 
type InterfaceCountersMap map[string]map[string]uint64
var interfaceCountersMap InterfaceCountersMap

Reply:
Done.